tests/manage/mcg/test_bucket_policy.py

```diff
@@ -311,8 +313,414 @@ def test_anonymous_read_only(self, mcg_obj, bucket_factory):
+        assert helpers.s3_put_bucket_versioning(
+            s3_obj=obc_obj, bucketname=obc_obj.bucket_name, status="Enabled"
+        ), "Failed: PutBucketVersioning"
+
+        logger.info(f'Verifying whether versioning is enabled on bucket: {obc_obj.bucket_name}')
+        assert helpers.s3_get_bucket_versioning(
+            s3_obj=obc_obj, bucketname=obc_obj.bucket_name
+        ), "Failed: GetBucketVersioning"
+
+        # Admin modifies the policy to all obc-account to write/read/delete versioned objects
+        bucket_policy_generated = gen_bucket_policy(
+            user_list=obc_obj.obc_account,
+            actions_list=object_version_action_list,
+            resources_list=[obc_obj.bucket_name, f'{obc_obj.bucket_name}/{"*"}']
+        )
+        bucket_policy = json.dumps(bucket_policy_generated)
+
+        logger.info(f'Creating bucket policy on bucket: {obc_obj.bucket_name} by Admin')
+        assert helpers.put_bucket_policy(mcg_obj, obc_obj.bucket_name, bucket_policy), "Failed: PutBucketPolicy"
+
+        # Getting Policy
+        logger.info(f'Getting bucket policy for bucket: {obc_obj.bucket_name}')
+        get_policy = helpers.get_bucket_policy(mcg_obj, obc_obj.bucket_name)
+        logger.info('Got bucket policy:%s' % get_policy['Policy'])
+
+        for key in range(5):
+            logger.info(f"Writing {key} version of {object_key}")
+            obj = helpers.s3_put_object(
+                s3_obj=obc_obj, bucketname=obc_obj.bucket_name,
+                object_key=object_key, data=data
+            )
+            object_versions.append(obj['VersionId'])
+
+        for version in object_versions:
+            logger.info(f"Reading version: {version} of {object_key}")
+            assert helpers.s3_get_object(
+                s3_obj=obc_obj, bucketname=obc_obj.bucket_name, object_key=object_key, versionid=version
+            ), f"Failed: To Read object {version}"
+            logger.info(f"Deleting version: {version} of {object_key}")
+            assert helpers.s3_delete_object(
+                s3_obj=obc_obj, bucketname=obc_obj.bucket_name, object_key=object_key, versionid=version
+            ), f"Failed: To Delete object with {version}"
+
+        bucket_policy_generated = gen_bucket_policy(
+            user_list=obc_obj.obc_account,
+            actions_list=['PutBucketVersioning'],
+            resources_list=[obc_obj.bucket_name]
+        )
+        bucket_policy = json.dumps(bucket_policy_generated)
+
+        logger.info(f'Creating bucket policy on bucket: {obc_obj.bucket_name} by Admin')
+        assert helpers.put_bucket_policy(mcg_obj, obc_obj.bucket_name, bucket_policy), "Failed: PutBucketPolicy"
+
+        # Getting Policy
+        logger.info(f'Getting bucket policy on bucket: {obc_obj.bucket_name}')
+        get_policy = helpers.get_bucket_policy(mcg_obj, obc_obj.bucket_name)
+        logger.info('Got bucket policy:%s' % get_policy['Policy'])
```

Comment:
why not f'Got bucket policy:{get_policy['Policy']}'

Reply:
ack